src/lib.rs

```diff
@@ -662,6 +662,33 @@ impl Shell {
         let cd = self.cwd.borrow();
         cd.join(p)
     }
+
+    /// Returns the environment variables set for this shell.
+    ///
+    /// The returned hash map contains a snapshot of the process’s environment variables at the time of this invocation. Modifications to environment variables afterwards will not be reflected in the returned iterator.
+    pub fn vars_os(&self) -> Vec<(OsString, OsString)> {
+        Vec::from_iter(self.env.borrow().iter().map(|(k, v)| (k.to_owned(), v.to_owned())))
+    }
+
+    /// Returns the environment variables set for this shell.
+    ///
+    /// The returned hash map contains a snapshot of the process’s environment
+    /// variables at the time of this invocation. Modifications to environment
+    /// variables afterwards will not be reflected in the returned iterator.
+    ///
+    /// # Panics
+    ///
+    /// While iterating, the returned iterator will panic if any key or value in
+    /// the environment is not valid unicode. If this is not desired, consider
+    /// using [`Shell::vars_os`](Self::vars_os).
+    pub fn vars(&self) -> Vec<(String, String)> {
```

Comment:
We should filter them out instead of panicking, I think that's also what the std one does.

Reply:
@lnicola Surprisingly: https://doc.rust-lang.org/stable/std/env/fn.vars.html#panics

Reply:
I guess filtering has the difficulty of hiding something the user might want to be aware of? Perhaps we ought to return a `Result` then?

Reply:
Oh, sorry, for some reason I thought it did that. It even made sense in a way 😀.